drivers/video/video_mcux_csi.c

```diff
@@ -114,23 +117,62 @@ static void __frame_done_cb(CSI_Type *base, csi_handle_t *handle, status_t statu
 	return;
 }
 
+#if defined(CONFIG_VIDEO_MCUX_MIPI_CSI2RX)
+K_HEAP_DEFINE(csi_heap, 1000);
```

Comment:
Would it not be possible to statically allocate the space for the CSI formats within the device data struct, and initialize them in `video_mcux_csi_set_fmt`? It doesn't seem like we need to use a heap for this single function call, versus statically allocated memory (since allocation is generally not desirable in some embedded applications)

Reply:
No, I think it's not possible. The formats caps come frome the camera sensor. We have no information about it until it is propagated from the sensor (via get_caps()).  Since it (video_format_cap) is a const struct pointer defined by the API, I have to declare a new struct to change it (unless **_we agree to de-const this struct_** in the video API but I don't think it is a good way).